tests/main/security-device-cgroups-serial-port/task.yaml

```diff
@@ -16,58 +16,60 @@ restore: |
+    NOMATCH "c 4:68 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
+    fi
 
-        echo "When a udev rule assigning the device to the snap is added"
-        content="SUBSYSTEM==\"tty\", KERNEL==\"ttyS4\", TAG+=\"snap_test-snapd-sh_sh\""
-        echo "$content" > /etc/udev/rules.d/70-snap.test-snapd-sh.rules
-        udevadm control --reload-rules
-        udevadm settle
-        udevadm trigger
-        udevadm settle
+    echo "When a udev rule assigning the device to the snap is added"
+    content="SUBSYSTEM==\"tty\", KERNEL==\"ttyS4\", TAG+=\"snap_test-snapd-sh_sh\""
+    echo "$content" > /etc/udev/rules.d/70-snap.test-snapd-sh.rules
+    udevadm control --reload-rules
+    udevadm settle
+    udevadm trigger
+    udevadm settle
 
-        echo "Then the device is shown as assigned to the snap"
-        udevadm info /dev/ttyS4 | MATCH "E: TAGS=.*snap_test-snapd-sh_sh"
-        # CURRENT_TAGS just available on systemd 247+
-        if [ "$(systemctl --version | awk '/systemd [0-9]+/ { print $2 }')" -ge 247 ]; then
-            udevadm info /dev/ttyS4 | MATCH "E: CURRENT_TAGS=.*snap_test-snapd-sh_sh"
-        fi
+    echo "Then the device is shown as assigned to the snap"
+    udevadm info /dev/ttyS4 | MATCH "E: TAGS=.*snap_test-snapd-sh_sh"
+    # CURRENT_TAGS just available on systemd 247+
+    if [ "$(systemctl --version | awk '/systemd [0-9]+/ { print $2 }')" -ge 247 ]; then
+        udevadm info /dev/ttyS4 | MATCH "E: CURRENT_TAGS=.*snap_test-snapd-sh_sh"
+    fi
 
-        echo "When a snap command is called"
-        test-snapd-sh.sh -c 'true'
+    echo "When a snap command is called"
+    test-snapd-sh.sh -c 'true'
 
-        echo "Then the device is shown in the snap device list"
-        MATCH "c 4:68 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
+    echo "Then the device is shown in the snap device list"
+    MATCH "c 4:68 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
 
-        echo "Once the snap is removed, the current tags are automatically removed"
-        snap remove test-snapd-sh
-        udevadm info /dev/ttyS4 | NOMATCH "E: CURRENT_TAGS=.*snap_test-snapd-sh_sh"
-        test ! -f /etc/udev/rules.d/70-snap.test-snapd-sh.rules
+    echo "Once the snap is removed, the current tags are automatically removed"
+    snap remove test-snapd-sh
+    udevadm info /dev/ttyS4 | NOMATCH "E: CURRENT_TAGS=.*snap_test-snapd-sh_sh"
+    test ! -f /etc/udev/rules.d/70-snap.test-snapd-sh.rules
 
-        # Reboot needed just on systemd 247+
-        if [ "$(systemctl --version | awk '/systemd [0-9]+/ { print $2 }')" -ge 247 ]; then
-            REBOOT
-        fi
+    if [ "$(systemctl --version | awk '/systemd [0-9]+/ { print $2 }')" -ge 247 ]; then
+        # with systemd versions 247+, the TAGS are sticky, but CURRENT_TAGS has
+        # been updated updated and checked
+        udevadm info /dev/ttyS4 | MATCH "E: TAGS=.*snap_test-snapd-sh_sh"
```

Comment:
Is it really useful to have this branch here? It kinda feels like here we are just checking that systemd's behavior on TAGS is consistent? Does it matter to us at all that systemd 247+ TAGS are sticky?

Reply:
I hope that tags do not become non sticky again with some random systemd/kernel release, until that time we trust and verify. FWIW, when all cgroup v2 lands, the branch will be executed on all distros, so we'll know sooner than the next Ubuntu release.

Reply:
okay, fair enough, thanks for clarifying